packages/site/cms-site/src/blocks/factories/BlocksBlock.tsx

```diff
@@ -31,9 +36,32 @@ export const BlocksBlock: React.FC<Props> = ({ supportedBlocks, data: { blocks }
 
                     return null;
                 }
-
-                return <React.Fragment key={block.key}>{blockFunction(block.props)}</React.Fragment>;
+                return (
+                    <React.Fragment key={block.key}>
+                        <ErrorBoundary
+                            blockType={block.type}
+                            fallback={process.env.NODE_ENV === "production" ? null : <ErrorFallback blockType={block.type} />}
+                            reportError={reportError}
+                        >
+                            {blockFunction(block.props)}
+                        </ErrorBoundary>
+                    </React.Fragment>
+                );
             })}
         </>
     );
 };
+
+function ErrorFallback({ blockType }: { blockType: string }) {
+    return <ErrorRoot>The following Block failed to render: {blockType}</ErrorRoot>;
+}
+
+const ErrorRoot = styled.div`
+    background-color: red;
+    color: white;
+    min-height: 500px;
```

Comment:
Is this min-height needed? Maybe add a padding instead?

Reply:
--> resolved because this will be removed (I'll throw an error instead of this red placeholder box)